testcases/kernel/syscalls/ipc/lib/libipc.c

```diff
@@ -150,6 +151,34 @@ int getuserid(char *user)
 
 	return (ent->pw_uid);
 }
+/*
+ *get_shm() - get a shared memory segment
+ */
+int get_shm(size_t size,int shmflg,int * shmid,void** shmaddr)
+{
+    int shmid_tmp;
+    key_t shmkey;
+
+    shmkey = getipckey();
+    if(shmkey < 0){
+        tst_brkm(TBROK,NULL,"getipckey() Failed to get a ipc key"
+                "key,error:%d",errno);
+    }
+    shmid_tmp = shmget(shmkey,size,shmflg);
+    if(shmid_tmp == -1){
+        tst_brkm(TBROK,NULL,"shmget() Failed to create a shared"
+                "memory,error:%d",errno);
+    }
+    *shmid = shmid_tmp;
+    *shmaddr = shmat(shmid_tmp,NULL,0);
+    if(*shmaddr == (void *) -1){
+        tst_brkm(TBROK,cleanup,"shmat() Failed to attach shared"
+                "memory,error:%d",errno);
+    }
+
+    return 0;
+
+}
 
```

Comment:
First of all, the libipc.c is being replaced with libnewipc.c, so adding more functionality to the old library is not acceptable any more.

Also there is no rationale why is this function needed at all and no subsequent patches that use this newly added functionality, which is another reason why this patch is not acceptable as it is. We do not add library function unless they are needed for something.

Strangely some of the tst_brkm() calls pass a cleanup function pointer, some does not, that looks like a bug.

And lastly the coding style is also wrong, but that is a minor problem. You can use checkpatch.pl from kernel git repo to check coding style violations.